v22.2/encryption.md

```diff
@@ -17,20 +17,96 @@ Cockroach determines which encryption algorithm to use based on the size of the
 | AES-192 | 192 bits (24 bytes) | 56 bytes |
 | AES-256 | 256 bits (32 bytes) | 64 bytes |
 
-Generating a key file can be done using the `cockroach` CLI:
+You can generate a store key for your cluster any of a several ways:
+
+
+### Using the `cockroach` CLI
 
 {% include_cached copy-clipboard.html %}
 ~~~ shell
 $ cockroach gen encryption-key -s 128 /path/to/my/aes-128.key
 ~~~
 
-Or the equivalent [openssl](https://www.openssl.org/docs/man1.1.1/man1/openssl.html) CLI command:
+### Using [OpenSSL](https://www.openssl.org/docs/man1.1.1/man1/openssl.html) CLI command
 
 {% include_cached copy-clipboard.html %}
 ~~~ shell
 $ openssl rand -out /path/to/my/aes-128.key 48
 ~~~
 
+### Using HashiCorp Vault
+
+First we need to enable the transit secret engine, which is going to handle the user creation for us.
+
+{% include_cached copy-clipboard.html %}
+~~~shell
+vault secrets enable transit
+~~~
+
+~~~txt
+
+~~~
+
+Next we need to create the exportable key. 
+
+
+{% include_cached copy-clipboard.html %}
+~~~shell
+$ vault write -f transit/keys/crdb-key -exportable=true 
+~~~
+
+~~~txt
+
+~~~
+
+And finally export it.
+
+{% include_cached copy-clipboard.html %}
+~~~shell
+vault read /transit/export/encryption-key/crdb-key/1
+
+~~~
+
+~~~txt
+Key     Value
+---     -----
+keys    map[1:6kkUAMq4EgIJKEmFxl97bQaBn0n9FL6FbRoUeCBtVhE=]
+name    crdb-key
+type    aes256-gcm96
+~~~
+
+But stop, since the exportable key is “just” 256 bits and cockroachDB has a special mechanic here. Let me quote our docs “The key file must contain random data making up the key ID (32 bytes) and the actual key (16, 24, or 32 bytes depending on the encryption algorithm).”
```

Comment:
This seems a little informal. It's not ideal to lead the reader into a mistake that they need to then recover from. I wonder if it's worth explaining this limitation / workaround earlier in the process.

Reply:
AFAIK there is no other way/order of doing things.